### PR TITLE
tests/lib/reset.sh: fix removing disabled snaps

### PR DESCRIPTION
**Pull request by anonymouse64**
This was actually not even parsing correctly, since each word in every line is
what we would loop over, not each line. Shell scripts, much like boulders, are
hard.

This is needed to make the snapd-failover, snapd-refresh-vs-services and 
snapd-refresh-vs-services-reboots tests more robust.

### CODE REVIEW

**Comment by anonymouse64** [comment]
This should hopefully fix the regression introduced by https://github.com/snapcore/snapd/pull/10517

**Review by stolowski — APPROVED** [review]
Nice catch! And it's tricky indeed, I had to play with the original loop and your change to get it ... Thanks!

File: tests/lib/reset.sh [hunk 1]
```diff
@@ -201,9 +201,9 @@ reset_all_snap() {
 # snapd 2.49.2 installed
 #
 
-for snListLine in $(snap list --all | grep disabled); do
-    name="$(echo "$snListLine" | awk '{print $1}')"
-    rev="$(echo "$snListLine" | awk '{print $3}')"
+snap list --all | grep disabled | while read -r line ; do
```
Comment (mardy) [inline tests/lib/reset.sh:204]:
```suggestion
snap list --all | grep disabled | while read -r name version revision restOfLine ; do
```

And you can get rid of the awk invocations.

**Review by mardy — APPROVED** [review]
Just a minor improvement (if you like) :-)

**Review by sergiocazzolato — APPROVED** [review]
Nice, thanks

File: tests/lib/reset.sh (hunk 1 above)
Comment (anonymouse64) [reply tests/lib/reset.sh]: looks good thanks for the suggestion!

**Pull request merged**